--- igraph.info	2008-07-11 07:10:31.000000000 -0500
+++ igraph.info.orig	2008-07-11 07:13:28.000000000 -0500
@@ -1,8 +1,8 @@
 This is igraph_reference_manual.info, produced by makeinfo version 4.11 from igraph.texi.
 
+INFO-DIR-SECTION Development
 START-INFO-DIR-ENTRY
-* igraph Reference Manual: (igraph_reference_manual).
-                                                  [MISSING TEXT]
+* Igraph Reference Manual: (igraph.info).    A library for graphs
 END-INFO-DIR-ENTRY
 
 
