--- igraph.info.orig	2019-08-16 10:35:48.540887736 +0200
+++ igraph.info	2019-08-16 10:41:57.388760592 +0200
@@ -2,8 +2,7 @@
 from igraph.texi.
 
 START-INFO-DIR-ENTRY
-* igraph Reference Manual: (igraph_reference_manual).
-                                                  [MISSING TEXT]
+* Igraph Reference Manual: (igraph.info).    A library for graphs, version @IGRAPH_VERSION@
 END-INFO-DIR-ENTRY
 
 
@@ -11,6 +10,7 @@
 
 igraph Reference Manual
 ***********************
+Version @IGRAPH_VERSION@
 
 * Menu:
 
