diff --git dgetv0.c dgetv0.c
index 36177bcc1..841602cfa 100644
--- dgetv0.c
+++ dgetv0.c
@@ -144,7 +144,7 @@ static doublereal c_b31 = -1.;
 {
     /* Initialized data */
 
-    static logical inits = TRUE_;
+    IGRAPH_F77_SAVE logical inits = TRUE_;
 
     /* System generated locals */
     integer v_dim1, v_offset, i__1;
@@ -157,29 +157,29 @@ static doublereal c_b31 = -1.;
     integer jj, nbx;
     extern doublereal igraphddot_(integer *, doublereal *, integer *, doublereal *,
 	    integer *);
-    static integer iter;
-    static logical orth;
+    IGRAPH_F77_SAVE integer iter;
+    IGRAPH_F77_SAVE logical orth;
     integer nopx;
     extern doublereal igraphdnrm2_(integer *, doublereal *, integer *);
-    static integer iseed[4];
+    IGRAPH_F77_SAVE integer iseed[4];
     extern /* Subroutine */ int igraphdgemv_(char *, integer *, integer *,
 	    doublereal *, doublereal *, integer *, doublereal *, integer *,
 	    doublereal *, doublereal *, integer *);
     integer idist;
     extern /* Subroutine */ int igraphdcopy_(integer *, doublereal *, integer *,
 	    doublereal *, integer *);
-    static logical first;
+    IGRAPH_F77_SAVE logical first;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen);
     integer mgetv0;
     real tgetv0;
-    static doublereal rnorm0;
+    IGRAPH_F77_SAVE doublereal rnorm0;
     extern /* Subroutine */ int igrapharscnd_(real *);
     integer logfil, ndigit;
     extern /* Subroutine */ int igraphdlarnv_(integer *, integer *, integer *,
 	    doublereal *);
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     real tmvopx;
 
 
diff --git dlaln2.c dlaln2.c
index f5d447b81..cecbc6ca2 100644
--- dlaln2.c
+++ dlaln2.c
@@ -236,15 +236,15 @@ f">
 {
     /* Initialized data */
 
-    static logical zswap[4] = { FALSE_,FALSE_,TRUE_,TRUE_ };
-    static logical rswap[4] = { FALSE_,TRUE_,FALSE_,TRUE_ };
-    static integer ipivot[16]	/* was [4][4] */ = { 1,2,3,4,2,1,4,3,3,4,1,2,
+    IGRAPH_F77_SAVE logical zswap[4] = { FALSE_,FALSE_,TRUE_,TRUE_ };
+    IGRAPH_F77_SAVE logical rswap[4] = { FALSE_,TRUE_,FALSE_,TRUE_ };
+    IGRAPH_F77_SAVE integer ipivot[16]	/* was [4][4] */ = { 1,2,3,4,2,1,4,3,3,4,1,2,
 	    4,3,2,1 };
 
     /* System generated locals */
     integer a_dim1, a_offset, b_dim1, b_offset, x_dim1, x_offset;
     doublereal d__1, d__2, d__3, d__4, d__5, d__6;
-    static doublereal equiv_0[4], equiv_1[4];
+    IGRAPH_F77_SAVE doublereal equiv_0[4], equiv_1[4];
 
     /* Local variables */
     integer j;
diff --git dlasy2.c dlasy2.c
index 1644a2e59..5f6d833a7 100644
--- dlasy2.c
+++ dlasy2.c
@@ -198,11 +198,11 @@ f">
 {
     /* Initialized data */
 
-    static integer locu12[4] = { 3,4,1,2 };
-    static integer locl21[4] = { 2,1,4,3 };
-    static integer locu22[4] = { 4,3,2,1 };
-    static logical xswpiv[4] = { FALSE_,FALSE_,TRUE_,TRUE_ };
-    static logical bswpiv[4] = { FALSE_,TRUE_,FALSE_,TRUE_ };
+    IGRAPH_F77_SAVE integer locu12[4] = { 3,4,1,2 };
+    IGRAPH_F77_SAVE integer locl21[4] = { 2,1,4,3 };
+    IGRAPH_F77_SAVE integer locu22[4] = { 4,3,2,1 };
+    IGRAPH_F77_SAVE logical xswpiv[4] = { FALSE_,FALSE_,TRUE_,TRUE_ };
+    IGRAPH_F77_SAVE logical bswpiv[4] = { FALSE_,TRUE_,FALSE_,TRUE_ };
 
     /* System generated locals */
     integer b_dim1, b_offset, tl_dim1, tl_offset, tr_dim1, tr_offset, x_dim1,
diff --git dnaitr.c dnaitr.c
index 7f224f29b..7c07a5574 100644
--- dnaitr.c
+++ dnaitr.c
@@ -236,7 +236,7 @@ static integer c__2 = 2;
 {
     /* Initialized data */
 
-    static logical first = TRUE_;
+    IGRAPH_F77_SAVE logical first = TRUE_;
 
     /* System generated locals */
     integer h_dim1, h_offset, v_dim1, v_offset, i__1, i__2;
@@ -247,24 +247,24 @@ static integer c__2 = 2;
 
     /* Local variables */
     integer i__;
-    static integer j;
+    IGRAPH_F77_SAVE integer j;
     real t0, t1, t2, t3, t4, t5;
     integer jj;
-    static integer ipj, irj;
+    IGRAPH_F77_SAVE integer ipj, irj;
     integer nbx;
-    static integer ivj;
-    static doublereal ulp;
+    IGRAPH_F77_SAVE integer ivj;
+    IGRAPH_F77_SAVE doublereal ulp;
     doublereal tst1;
     extern doublereal igraphddot_(integer *, doublereal *, integer *, doublereal *,
 	    integer *);
-    static integer ierr, iter;
-    static doublereal unfl, ovfl;
+    IGRAPH_F77_SAVE integer ierr, iter;
+    IGRAPH_F77_SAVE doublereal unfl, ovfl;
     integer nopx;
-    static integer itry;
+    IGRAPH_F77_SAVE integer itry;
     extern doublereal igraphdnrm2_(integer *, doublereal *, integer *);
     doublereal temp1;
-    static logical orth1, orth2, step3, step4;
-    static doublereal betaj;
+    IGRAPH_F77_SAVE logical orth1, orth2, step3, step4;
+    IGRAPH_F77_SAVE doublereal betaj;
     extern /* Subroutine */ int igraphdscal_(integer *, doublereal *, doublereal *,
 	    integer *), igraphdgemv_(char *, integer *, integer *, doublereal *,
 	    doublereal *, integer *, doublereal *, integer *, doublereal *,
@@ -279,12 +279,12 @@ static integer c__2 = 2;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen);
-    static doublereal wnorm;
+    IGRAPH_F77_SAVE doublereal wnorm;
     extern /* Subroutine */ int igraphivout_(integer *, integer *, integer *,
 	    integer *, char *, ftnlen), igraphdgetv0_(integer *, char *, integer *,
 	    logical *, integer *, integer *, doublereal *, integer *,
 	    doublereal *, doublereal *, integer *, doublereal *, integer *), igraphdlabad_(doublereal *, doublereal *);
-    static doublereal rnorm1;
+    IGRAPH_F77_SAVE doublereal rnorm1;
     extern doublereal igraphdlamch_(char *);
     extern /* Subroutine */ int igraphdlascl_(char *, integer *, integer *,
 	    doublereal *, doublereal *, integer *, integer *, doublereal *,
@@ -294,10 +294,10 @@ static integer c__2 = 2;
     extern /* Subroutine */ int igrapharscnd_(real *);
     integer logfil, ndigit, nitref, mnaitr;
     real titref, tnaitr;
-    static integer msglvl;
-    static doublereal smlnum;
+    IGRAPH_F77_SAVE integer msglvl;
+    IGRAPH_F77_SAVE doublereal smlnum;
     integer nrorth;
-    static logical rstart;
+    IGRAPH_F77_SAVE logical rstart;
     integer nrstrt;
     real tmvopx;
 
diff --git dnapps.c dnapps.c
index 094579725..732236091 100644
--- dnapps.c
+++ dnapps.c
@@ -168,7 +168,7 @@ static doublereal c_b43 = -1.;
 {
     /* Initialized data */
 
-    static logical first = TRUE_;
+    IGRAPH_F77_SAVE logical first = TRUE_;
 
     /* System generated locals */
     integer h_dim1, h_offset, v_dim1, v_offset, q_dim1, q_offset, i__1, i__2,
@@ -183,10 +183,10 @@ static doublereal c_b43 = -1.;
     doublereal h11, h12, h21, h22, h32;
     integer jj, ir, nr;
     doublereal tau;
-    static doublereal ulp;
+    IGRAPH_F77_SAVE doublereal ulp;
     doublereal tst1;
     integer iend;
-    static doublereal unfl, ovfl;
+    IGRAPH_F77_SAVE doublereal unfl, ovfl;
     extern /* Subroutine */ int igraphdscal_(integer *, doublereal *, doublereal *,
 	    integer *), igraphdlarf_(char *, integer *, integer *, doublereal *,
 	    integer *, doublereal *, doublereal *, integer *, doublereal *);
@@ -218,7 +218,7 @@ static doublereal c_b43 = -1.;
     integer mnapps, msglvl;
     real tnapps;
     integer istart;
-    static doublereal smlnum;
+    IGRAPH_F77_SAVE doublereal smlnum;
     integer kplusp;
 
 
diff --git dnaup2.c dnaup2.c
index 001a395ca..46c6049d9 100644
--- dnaup2.c
+++ dnaup2.c
@@ -216,24 +216,24 @@ static integer c__2 = 2;
     integer j;
     real t0, t1, t2, t3;
     integer kp[4];
-    static integer np0;
+    IGRAPH_F77_SAVE integer np0;
     integer nbx;
-    static integer nev0;
+    IGRAPH_F77_SAVE integer nev0;
     extern doublereal igraphddot_(integer *, doublereal *, integer *, doublereal *,
 	    integer *);
-    static doublereal eps23;
+    IGRAPH_F77_SAVE doublereal eps23;
     integer ierr;
-    static integer iter;
+    IGRAPH_F77_SAVE integer iter;
     doublereal temp;
     extern doublereal igraphdnrm2_(integer *, doublereal *, integer *);
-    static logical getv0, cnorm;
+    IGRAPH_F77_SAVE logical getv0, cnorm;
     extern /* Subroutine */ int igraphdcopy_(integer *, doublereal *, integer *,
 	    doublereal *, integer *);
-    static integer nconv;
+    IGRAPH_F77_SAVE integer nconv;
     extern /* Subroutine */ int igraphdmout_(integer *, integer *, integer *,
 	    doublereal *, integer *, integer *, char *, ftnlen);
-    static logical initv;
-    static doublereal rnorm;
+    IGRAPH_F77_SAVE logical initv;
+    IGRAPH_F77_SAVE doublereal rnorm;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen), igraphivout_(integer *, integer *, integer *
@@ -247,14 +247,14 @@ static integer c__2 = 2;
     extern /* Subroutine */ int igraphdneigh_(doublereal *, integer *, doublereal *,
 	     integer *, doublereal *, doublereal *, doublereal *, doublereal *
 	    , integer *, doublereal *, integer *);
-    static integer nevbef;
+    IGRAPH_F77_SAVE integer nevbef;
     extern /* Subroutine */ int igrapharscnd_(real *);
     integer logfil, ndigit;
     extern /* Subroutine */ int igraphdnaitr_(integer *, char *, integer *, integer
 	    *, integer *, integer *, doublereal *, doublereal *, doublereal *,
 	     integer *, doublereal *, integer *, integer *, doublereal *,
 	    integer *);
-    static logical update;
+    IGRAPH_F77_SAVE logical update;
     extern /* Subroutine */ int igraphdngets_(integer *, char *, integer *, integer
 	    *, doublereal *, doublereal *, doublereal *, doublereal *,
 	    doublereal *), igraphdnapps_(integer *, integer *, integer *,
@@ -263,11 +263,11 @@ static integer c__2 = 2;
 	    doublereal *), igraphdnconv_(integer *, doublereal *, doublereal *,
 	    doublereal *, doublereal *, integer *), igraphdsortc_(char *, logical *,
 	     integer *, doublereal *, doublereal *, doublereal *);
-    static logical ushift;
+    IGRAPH_F77_SAVE logical ushift;
     char wprime[2];
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     integer nptemp;
-    static integer numcnv, kplusp;
+    IGRAPH_F77_SAVE integer numcnv, kplusp;
 
 
 /*     %----------------------------------------------------%
diff --git dnaupd.c dnaupd.c
index e1fceaaaf..03feb6aa2 100644
--- dnaupd.c
+++ dnaupd.c
@@ -464,18 +464,18 @@ static integer c__1 = 1;
     /* Local variables */
     integer j;
     real t0, t1;
-    static integer nb, ih, iq, np, iw, ldh, ldq;
+    IGRAPH_F77_SAVE integer nb, ih, iq, np, iw, ldh, ldq;
     integer nbx;
-    static integer nev0, mode;
+    IGRAPH_F77_SAVE integer nev0, mode;
     integer ierr;
-    static integer iupd, next;
+    IGRAPH_F77_SAVE integer iupd, next;
     integer nopx;
     real trvec, tmvbx;
-    static integer ritzi;
+    IGRAPH_F77_SAVE integer ritzi;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen), igraphivout_(integer *, integer *, integer *
 	    , integer *, char *, ftnlen);
-    static integer ritzr;
+    IGRAPH_F77_SAVE integer ritzr;
     extern /* Subroutine */ int igraphdnaup2_(integer *, char *, integer *, char *,
 	    integer *, integer *, doublereal *, doublereal *, integer *,
 	    integer *, integer *, integer *, doublereal *, integer *,
@@ -488,15 +488,15 @@ static integer c__1 = 1;
     integer logfil, ndigit;
     real tneigh;
     integer mnaupd;
-    static integer ishift;
+    IGRAPH_F77_SAVE integer ishift;
     integer nitref;
-    static integer bounds;
+    IGRAPH_F77_SAVE integer bounds;
     real tnaupd;
     extern /* Subroutine */ int igraphdstatn_(void);
     real titref, tnaitr;
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     real tngets, tnapps, tnconv;
-    static integer mxiter;
+    IGRAPH_F77_SAVE integer mxiter;
     integer nrorth, nrstrt;
     real tmvopx;
 
diff --git dsaitr.c dsaitr.c
index 9a2da9ded..396ec741c 100644
--- dsaitr.c
+++ dsaitr.c
@@ -231,7 +231,7 @@ static integer c__2 = 2;
 {
     /* Initialized data */
 
-    static logical first = TRUE_;
+    IGRAPH_F77_SAVE logical first = TRUE_;
 
     /* System generated locals */
     integer h_dim1, h_offset, v_dim1, v_offset, i__1;
@@ -241,20 +241,20 @@ static integer c__2 = 2;
 
     /* Local variables */
     integer i__;
-    static integer j;
+    IGRAPH_F77_SAVE integer j;
     real t0, t1, t2, t3, t4, t5;
     integer jj;
-    static integer ipj, irj;
+    IGRAPH_F77_SAVE integer ipj, irj;
     integer nbx;
-    static integer ivj;
+    IGRAPH_F77_SAVE integer ivj;
     extern doublereal igraphddot_(integer *, doublereal *, integer *, doublereal *,
 	    integer *);
-    static integer ierr, iter;
+    IGRAPH_F77_SAVE integer ierr, iter;
     integer nopx;
-    static integer itry;
+    IGRAPH_F77_SAVE integer itry;
     extern doublereal igraphdnrm2_(integer *, doublereal *, integer *);
     doublereal temp1;
-    static logical orth1, orth2, step3, step4;
+    IGRAPH_F77_SAVE logical orth1, orth2, step3, step4;
     extern /* Subroutine */ int igraphdscal_(integer *, doublereal *, doublereal *,
 	    integer *), igraphdgemv_(char *, integer *, integer *, doublereal *,
 	    doublereal *, integer *, doublereal *, integer *, doublereal *,
@@ -266,25 +266,25 @@ static integer c__2 = 2;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen);
-    static doublereal wnorm;
+    IGRAPH_F77_SAVE doublereal wnorm;
     extern /* Subroutine */ int igraphivout_(integer *, integer *, integer *,
 	    integer *, char *, ftnlen), igraphdgetv0_(integer *, char *, integer *,
 	    logical *, integer *, integer *, doublereal *, integer *,
 	    doublereal *, doublereal *, integer *, doublereal *, integer *);
-    static doublereal rnorm1;
+    IGRAPH_F77_SAVE doublereal rnorm1;
     extern doublereal igraphdlamch_(char *);
     extern /* Subroutine */ int igraphdlascl_(char *, integer *, integer *,
 	    doublereal *, doublereal *, integer *, integer *, doublereal *,
 	    integer *, integer *), igrapharscnd_(real *);
     integer logfil;
-    static doublereal safmin;
+    IGRAPH_F77_SAVE doublereal safmin;
     integer ndigit, nitref;
     real titref;
     integer msaitr;
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     real tsaitr;
     integer nrorth;
-    static logical rstart;
+    IGRAPH_F77_SAVE logical rstart;
     integer nrstrt;
     real tmvopx;
 
diff --git dsapps.c dsapps.c
index 144d29d52..437eb3698 100644
--- dsapps.c
+++ dsapps.c
@@ -156,7 +156,7 @@ static doublereal c_b20 = -1.;
 {
     /* Initialized data */
 
-    static logical first = TRUE_;
+    IGRAPH_F77_SAVE logical first = TRUE_;
 
     /* System generated locals */
     integer h_dim1, h_offset, q_dim1, q_offset, v_dim1, v_offset, i__1, i__2,
@@ -185,7 +185,7 @@ static doublereal c_b20 = -1.;
 	    doublereal *, doublereal *, doublereal *, doublereal *), igraphdlaset_(
 	    char *, integer *, integer *, doublereal *, doublereal *,
 	    doublereal *, integer *);
-    static doublereal epsmch;
+    IGRAPH_F77_SAVE doublereal epsmch;
     integer logfil, ndigit, msapps, msglvl, istart;
     real tsapps;
     integer kplusp;
diff --git dsaup2.c dsaup2.c
index 51a83456c..95cee5189 100644
--- dsaup2.c
+++ dsaup2.c
@@ -221,26 +221,26 @@ static integer c__2 = 2;
     integer j;
     real t0, t1, t2, t3;
     integer kp[3];
-    static integer np0;
+    IGRAPH_F77_SAVE integer np0;
     integer nbx;
-    static integer nev0;
+    IGRAPH_F77_SAVE integer nev0;
     extern doublereal igraphddot_(integer *, doublereal *, integer *, doublereal *,
 	    integer *);
-    static doublereal eps23;
+    IGRAPH_F77_SAVE doublereal eps23;
     integer ierr;
-    static integer iter;
+    IGRAPH_F77_SAVE integer iter;
     doublereal temp;
     integer nevd2;
     extern doublereal igraphdnrm2_(integer *, doublereal *, integer *);
-    static logical getv0;
+    IGRAPH_F77_SAVE logical getv0;
     integer nevm2;
-    static logical cnorm;
+    IGRAPH_F77_SAVE logical cnorm;
     extern /* Subroutine */ int igraphdcopy_(integer *, doublereal *, integer *,
 	    doublereal *, integer *), igraphdswap_(integer *, doublereal *, integer
 	    *, doublereal *, integer *);
-    static integer nconv;
-    static logical initv;
-    static doublereal rnorm;
+    IGRAPH_F77_SAVE integer nconv;
+    IGRAPH_F77_SAVE logical initv;
+    IGRAPH_F77_SAVE doublereal rnorm;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen), igraphivout_(integer *, integer *, integer *
@@ -255,7 +255,7 @@ static integer c__2 = 2;
     integer logfil, ndigit;
     extern /* Subroutine */ int igraphdseigt_(doublereal *, integer *, doublereal *,
 	     integer *, doublereal *, doublereal *, doublereal *, integer *);
-    static logical update;
+    IGRAPH_F77_SAVE logical update;
     extern /* Subroutine */ int igraphdsaitr_(integer *, char *, integer *, integer
 	    *, integer *, integer *, doublereal *, doublereal *, doublereal *,
 	     integer *, doublereal *, integer *, integer *, doublereal *,
@@ -265,13 +265,13 @@ static integer c__2 = 2;
 	    integer *, doublereal *, integer *, doublereal *, doublereal *,
 	    integer *, doublereal *), igraphdsconv_(integer *, doublereal *,
 	    doublereal *, doublereal *, integer *);
-    static logical ushift;
+    IGRAPH_F77_SAVE logical ushift;
     char wprime[2];
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     integer nptemp;
     extern /* Subroutine */ int igraphdsortr_(char *, logical *, integer *,
 	    doublereal *, doublereal *);
-    static integer kplusp;
+    IGRAPH_F77_SAVE integer kplusp;
 
 
 /*     %----------------------------------------------------%
diff --git dsaupd.c dsaupd.c
index 8211fc293..b78da9b87 100644
--- dsaupd.c
+++ dsaupd.c
@@ -465,11 +465,11 @@ static integer c__1 = 1;
     /* Local variables */
     integer j;
     real t0, t1;
-    static integer nb, ih, iq, np, iw, ldh, ldq;
+    IGRAPH_F77_SAVE integer nb, ih, iq, np, iw, ldh, ldq;
     integer nbx;
-    static integer nev0, mode, ierr, iupd, next;
+    IGRAPH_F77_SAVE integer nev0, mode, ierr, iupd, next;
     integer nopx;
-    static integer ritz;
+    IGRAPH_F77_SAVE integer ritz;
     real tmvbx;
     extern /* Subroutine */ int igraphdvout_(integer *, integer *, doublereal *,
 	    integer *, char *, ftnlen), igraphivout_(integer *, integer *, integer *
@@ -483,14 +483,14 @@ static integer c__1 = 1;
     extern doublereal igraphdlamch_(char *);
     extern /* Subroutine */ int igrapharscnd_(real *);
     integer logfil, ndigit;
-    static integer ishift;
+    IGRAPH_F77_SAVE integer ishift;
     integer nitref, msaupd;
-    static integer bounds;
+    IGRAPH_F77_SAVE integer bounds;
     real titref, tseigt, tsaupd;
     extern /* Subroutine */ int igraphdstats_(void);
-    static integer msglvl;
+    IGRAPH_F77_SAVE integer msglvl;
     real tsaitr;
-    static integer mxiter;
+    IGRAPH_F77_SAVE integer mxiter;
     real tsgets, tsapps;
     integer nrorth;
     real tsconv;
